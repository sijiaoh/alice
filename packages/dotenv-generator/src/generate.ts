import fs from 'fs';
import { EOL } from 'os';
import path from 'path';
import dotenv from 'dotenv';

export const generate = (env = process.env.NODE_ENV) => {
  const base = dotenv.parse(fs.readFileSync(path.join(process.cwd(), '.env')));
  const overwritePath = path.join(process.cwd(), `.env${env ? `.${env}` : ''}`);
  const overwrite = dotenv.parse(fs.readFileSync(overwritePath));

  Object.keys(base).forEach((key) => {
    if (overwrite[key]) base[key] = overwrite[key];
    else if (!base[key])
      throw new Error(`${key} must set in ${overwritePath}.`);
  });

  return (
    [
      '// Generated by dotenv-generator.',
      `export const env = ${JSON.stringify(base, null, 2)};`,
    ].join(EOL) + EOL
  );
};
