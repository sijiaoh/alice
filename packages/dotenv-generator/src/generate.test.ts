import { generate } from './generate';

describe('generate', () => {
  describe('with legal env', () => {
    it('generate env.ts', () => {
      expect(generate()).toMatchSnapshot();
    });
  });
  describe('with illegal env', () => {
    it('throw error', () => {
      expect(() => {
        generate('test2');
      }).toThrow();
    });
  });
});
