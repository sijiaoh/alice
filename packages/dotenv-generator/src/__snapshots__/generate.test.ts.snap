// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate with legal env generate env.ts. 1`] = `
"// Generated by dotenv-generator.
export const env = {
  \\"A\\": \\"a\\",
  \\"B\\": \\"bb\\"
};
"
`;
